src/core/mempool.rs

```diff
@@ -1260,6 +1301,25 @@ impl MemPoolDB {
         Ok(updated)
     }
 
+    /// Helper method to record nonces to a retry-buffer.
+    /// This is needed for when we try to write-through a new (address, nonce) pair to the on-disk
+    /// `nonces` cache, but the write fails due to lock contention from another thread.  The
+    /// retry-buffer will be used to later store this data in a single transaction.
+    fn save_nonce_for_retry(
+        retry_store: &mut HashMap<StacksAddress, u64>,
+        max_size: u64,
+        addr: StacksAddress,
+        new_nonce: u64,
+    ) {
+        if (retry_store.len() as u64) < max_size {
+            if let Some(nonce) = retry_store.get_mut(&addr) {
+                *nonce = cmp::max(new_nonce, *nonce);
```

Comment:
In what scenario would this `max` be necessary? Couldn't we just simplify and overwrite what's there? I don't see any reason that a lower nonce gets passed here.